---
title: "Aneuploidy_Analyses_DDX3Feedback_SR_2022"
output: html_document
date: "2022-11-09"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
#Files from Sept 2022 for aneuploidies and structural variants
load(file="/lab/solexa_page/shruthir/Adrianna/txi_aneuploidy.rda")
allTPMs = as.data.frame(txi_aneuploidy$abundance)
DDX3_metadata = read.delim("/lab/solexa_page/shruthir/Adrianna/metadata_20210811.txt")
load(file = "/lab/solexa_page/shruthir/Adrianna/txi_lcl_strucvar.rda" )
lclstructpms = as.data.frame(txi_lcl_strucvar$abundance)

```

```{r}
#Extract expressed X-Y pairs 
lcl_struc_DDX3 = lclstructpms[c("DDX3X","DDX3Y","ZFX","ZFY","KDM6A","UTY","KDM5C","KDM5D","RPS4X","RPS4Y1","USP9X","USP9Y","PRKX","PRKY"),]
all_TPM_DDX3 = allTPMs[c("DDX3X","DDX3Y","ZFX","ZFY","KDM6A","UTY","KDM5C","KDM5D","RPS4X","RPS4Y1","USP9X","USP9Y","PRKX","PRKY"),]

```

```{r}
#Transpose data frames
all_TPM_DDX3 = as.data.frame(t(all_TPM_DDX3))
lcl_struc_DDX3 = as.data.frame(t(lcl_struc_DDX3))
```

```{r}
# Rename Metadata Rows After Sample Name, and Add Sample Column to Cell Line Dataframes
row.names(DDX3_metadata) = DDX3_metadata$sample
lcl_struc_DDX3$sample = row.names(lcl_struc_DDX3)
all_TPM_DDX3$sample = row.names(all_TPM_DDX3)
```

```{r}
# Separate Fibroblast and LCL Metadata
DDX3_metadata_fib = DDX3_metadata[which(DDX3_metadata$cell_type == "Fib"),]
DDX3_metadata_lcl = DDX3_metadata[which(DDX3_metadata$cell_type == "LCL"),]

```

```{r}
#Merge metadata and dataframes and split into fib and LCLs
lcl_struc_merge = merge(DDX3_metadata,lcl_struc_DDX3)
all_TPM_DDX3_merge = merge(DDX3_metadata, all_TPM_DDX3)

all_TPM_fib = all_TPM_DDX3_merge[which(all_TPM_DDX3_merge$cell_type == "Fib"),]
all_TPM_lcl = all_TPM_DDX3_merge[which(all_TPM_DDX3_merge$cell_type == "LCL"),]
```

```{r}
#Extract Which Samples Have Exactly 1 X
One_X_fib = all_TPM_fib[which(all_TPM_fib$x_count == 1),]
One_X_lcl = all_TPM_lcl[which(all_TPM_lcl$x_count == 1),]
```

```{r}
#Extract Which Samples Have Exactly 1 Y
One_Y_fib = all_TPM_fib[which(all_TPM_fib$y_count == 1), ]
One_Y_lcl = all_TPM_lcl[which(all_TPM_lcl$y_count == 1),]
```


```{r}
# Plot 1 X Chromosome DDX3X TPM Against Y-count scatter, and vice versa with Pearson correlation
library(ggplot2)
library(ggpubr)
library(ggbreak)
One_Y_l = data.frame(One_Y_lcl$x_count, One_Y_lcl$DDX3Y,One_Y_lcl$DDX3X)
One_X_l = data.frame(One_X_lcl$y_count, One_X_lcl$DDX3Y,One_X_lcl$DDX3X)
One_Y_f = data.frame(One_Y_fib$x_count, One_Y_fib$DDX3Y,One_Y_fib$DDX3X)
One_X_f = data.frame(One_X_fib$y_count, One_X_fib$DDX3Y,One_X_fib$DDX3X)


#Combine DDX3X and DDX3Y on same plot, dot plot plus line without SE, statistics in legend
#one Y lCL needs scale break 

oneylcl= ggplot() +
  geom_point(data = One_Y_l, aes(x =One_Y_lcl.x_count, y =One_Y_lcl.DDX3X), color = 'orange',position = position_jitter(w = 0.1, h = 0.1)) +
  geom_smooth(data = One_Y_l, aes(x =One_Y_lcl.x_count, y =One_Y_lcl.DDX3X), color = 'orange',method = 'lm',se=FALSE) +
  geom_point(data = One_Y_l, aes(x =One_Y_lcl.x_count, y =One_Y_lcl.DDX3Y), color = 'purple',position = position_jitter(w = 0.1, h = 0.1)) +
  geom_smooth(data = One_Y_l, aes(x =One_Y_lcl.x_count, y =One_Y_lcl.DDX3Y), color = 'purple',method ='lm',se=FALSE) +
  xlab("Number of X Chromosomes") +
  ylab("TPM") +
  theme_classic()
  #scale_y_break(c(40, 110))
ggsave(plot = oneylcl, width = 3.5, height = 3, dpi = 300, filename = "/lab/solexa_page/shruthir/Paperfigures/Aneuploidy/oneylcl.pdf")
   
```

```{r}
#one X lcl
onexlcl= ggplot() +
  geom_point(data = One_X_l, aes(x =One_X_lcl.y_count, y =One_X_lcl.DDX3X), color = 'orange',position = position_jitter(w = 0.2, h = 0.0)) +
  geom_smooth(data = One_X_l, aes(x =One_X_lcl.y_count, y =One_X_lcl.DDX3X), color = 'orange',method = 'lm',se=FALSE) +
  geom_point(data = One_X_l, aes(x =One_X_lcl.y_count, y =One_X_lcl.DDX3Y), color = 'purple',position = position_jitter(w = 0.2, h = 0.0)) +
  geom_smooth(data = One_X_l, aes(x =One_X_lcl.y_count, y =One_X_lcl.DDX3Y), color = 'purple',method ='lm',se=FALSE) +
  xlab("Number of Y Chromosomes") +
  ylab("TPM") +
  theme_classic()
  #scale_y_break(c(40, 110))
ggsave(plot = onexlcl, width = 3.5, height = 3, dpi = 300, filename = "/lab/solexa_page/shruthir/Paperfigures/Aneuploidy/onexlcl.pdf")
```

```{r}
#one Y fib 
oneyfib= ggplot() +
  geom_point(data = One_Y_f, aes(x =One_Y_fib.x_count, y =One_Y_fib.DDX3X), color = 'orange',position = position_jitter(w = 0.1, h = 0.1)) +
  geom_smooth(data = One_Y_f, aes(x =One_Y_fib.x_count, y =One_Y_fib.DDX3X), color = 'orange',method = 'lm',se=FALSE) +
  geom_point(data = One_Y_f, aes(x =One_Y_fib.x_count, y =One_Y_fib.DDX3Y), color = 'purple',position = position_jitter(w = 0.1, h = 0.1)) +
  geom_smooth(data = One_Y_f, aes(x =One_Y_fib.x_count, y =One_Y_fib.DDX3Y), color = 'purple',method ='lm',se=FALSE) +
  xlab("Number of X Chromosomes") +
  ylab("TPM") +
  theme_classic() 
  #scale_y_break(c(50, 95))
ggsave(plot = oneyfib, width = 3.5, height = 3, dpi = 300, filename = "/lab/solexa_page/shruthir/Paperfigures/Aneuploidy/oneyfib.pdf")
```

```{r}
#one X fib
onexfib= ggplot() +
  geom_point(data = One_X_f, aes(x =One_X_fib.y_count, y =One_X_fib.DDX3X), color = 'orange',position = position_jitter(w = 0.1, h = 0.1)) +
  geom_smooth(data = One_X_f, aes(x =One_X_fib.y_count, y =One_X_fib.DDX3X), color = 'orange',method = 'lm',se=FALSE) +
  geom_point(data = One_X_f, aes(x =One_X_fib.y_count, y =One_X_fib.DDX3Y), color = 'purple',position = position_jitter(w = 0.1, h = 0.1)) +
  geom_smooth(data = One_X_f, aes(x =One_X_fib.y_count, y =One_X_fib.DDX3Y), color = 'purple',method ='lm',se=FALSE) +
  xlab("Number of Y Chromosomes") +
  ylab("TPM") +
  theme_classic()
  #scale_y_break(c(40, 110))
ggsave(plot = onexfib, width = 3.5, height = 3, dpi = 300, filename = "/lab/solexa_page/shruthir/Paperfigures/Aneuploidy/onexfib.pdf")
```

```{r}
#Plot DDX3X and DDX3Y with Chr 21 response
#First load txi and extract TPM
load(file="/lab/solexa_page/shruthir/Adrianna/txi_lcl_t21.rda")
write.table(txi_lcl_t21$abundance, file="/lab/solexa_page/shruthir/Adrianna/txi_LCL21TPM.txt", sep="\t", quote=F,col.names = NA)

#Start HERE
options(scipen = 999)
lcl21TPM = read.delim("/lab/solexa_page/shruthir/Adrianna/txi_LCL21TPM.txt",check.names = FALSE)
row.names(lcl21TPM) = lcl21TPM$Gene
#Read in metadata
metadata21 = read.delim("/lab/solexa_page/shruthir/Adrianna/metadata_lcl_trisomy21.txt")
#Limit to relevant genes, restructure, Add row, name, merge with metadata 
lcl21 = lcl21TPM[c("DDX3X","DDX3Y","ZFX","ZFY","KDM6A","UTY","KDM5C","KDM5D","RPS4X","RPS4Y1","USP9X","USP9Y","PRKX","PRKY"),]
lcl21t = as.data.frame(t(lcl21))
#Remove gene row
lcl21t = lcl21t[-c(1), ]
#Same column
row.names(metadata21) = metadata21$sample
lcl21t$sample = row.names(lcl21t)
#Merge based on sample column
library(dplyr)
lcl21merge = merge(lcl21t,metadata21)
```

```{r}
#Make specific data frame and plot
#library(ggplot)
lcl21DDX3 = data.frame(lcl21merge$Chr21_count, lcl21merge$DDX3Y,lcl21merge$DDX3X)
lcl21DDX3 = lcl21DDX3[-c(17),]
options(scipen = 999)

lcl21 = ggplot() +
  geom_point(data = lcl21DDX3, aes(x =lcl21merge.Chr21_count, y =lcl21merge.DDX3X), color = 'orange',position = position_jitter(w = 0.1, h = 0.1)) +
  #geom_smooth(data = lcl21DDX3, aes(x =lcl21merge.Chr21_count, y =lcl21merge.DDX3X), color = 'orange',method = 'lm',se=FALSE) +
  geom_point(data = lcl21DDX3, aes(x =lcl21merge.Chr21_count, y =lcl21merge.DDX3Y), color = 'purple',position = position_jitter(w = 0.1, h = 0.1)) +
  scale_x_discrete(breaks = c(2,3))+
  scale_y_discrete(breaks = seq(0, 150, 20))+
  #geom_smooth(data = lcl21DDX3, aes(x =lcl21merge.Chr21_count, y =lcl21merge.DDX3Y), color = 'purple',method ='lm',se=FALSE) +
  xlab("Chromosome 21 Count") +
  ylab("TPM") +
  theme_classic()
ggsave(plot = lcl21, filename = "/lab/solexa_page/shruthir/Paperfigures/Aneuploidy/Chr21.pdf")
   

```


```{r}
#Replicate for other pairs. Define function that graphs all X-Y pairs. Define list of X-Y pair genes
#Run for all 
makegraph = function(dataframe) {
   graph = ggplot() +
    geom_point(data = dataframe, aes(x =dataframe[,1], y =dataframe[,2]), color = 'orange') +
    geom_smooth(data = dataframe, aes(x =dataframe[,1], y =dataframe[,2]), color = 'orange',method = 'lm',se=FALSE) +
    geom_point(data = dataframe, aes(x =dataframe[,1], y =dataframe[,3]), color = 'purple') +
    geom_smooth(data = dataframe, aes(x =dataframe[,1], y =dataframe[,3]), color = 'purple',method = 'lm',se=FALSE) +
    #xlab("Number of Y Chromosomes") +
    #ylab("TPM") +
    theme_classic()
    ggsave(graph,filename = paste0(dataframe,".png"))
}
```


```{r}
#Loop through all dataframes
listofframes = read.delim("/lab/solexa_page/shruthir/Paperfigures/Aneuploidy/dataframelist.txt", header = FALSE)
testlist = c("One_X_zfxl","One_Y_zfxl")
```



```{r}
#Create variable for summed XY pairs for all genes
all_TPM_fib$DDX3 = all_TPM_fib$DDX3X + all_TPM_fib$DDX3Y
all_TPM_lcl$DDX3 = all_TPM_lcl$DDX3X + all_TPM_lcl$DDX3Y

all_TPM_fib$ZF = all_TPM_fib$ZFX + all_TPM_fib$ZFY
all_TPM_lcl$ZF = all_TPM_lcl$ZFX + all_TPM_lcl$ZFY

all_TPM_fib$KDM = all_TPM_fib$KDM6A + all_TPM_fib$UTY
all_TPM_lcl$KDM = all_TPM_lcl$KDM6A + all_TPM_lcl$UTY

```

```{r}
#make new dataframe
lclsumanalysis = all_TPM_lcl[,c("sample","chrom_count","DDX3","ZF","KDM")]
```

```{r}
write.table(lclsumanalysis, file = "/lab/solexa_page/shruthir/Adrianna/lclsumanalysis.txt", sep="\t", quote=F, col.names=NA)
```



```{r}
#Plot total levels against chromosome count for lcl DDX3
total_l = data.frame(all_TPM_lcl$chrom_count, all_TPM_lcl$DDX3)
ggplot(total_l, aes(x =all_TPM_lcl.chrom_count, y =all_TPM_lcl.DDX3)) +
  geom_point(color = "black") + 
  labs(title = "DDX3 expression vs sex chromosome in LCLs") +
  #stat_cor(method = "pearson") +
  xlab("Number of Sex Chromosomes") +
  ylab("Total DDX3X + DDX3Y TPM") +
  geom_smooth(color='grey', se=FALSE, method = lm) +
  theme_classic()
```


```{r}
#Plot total levels against chromosome count for lcl ZFX
total_ZFXl = data.frame(all_TPM_lcl$chrom_count, all_TPM_lcl$ZF)
ggplot(total_ZFXl, aes(x =all_TPM_lcl.chrom_count, y =all_TPM_lcl.ZF)) +
  geom_point(color = "black") + 
  labs(title = "ZFX + ZFYexpression vs sex chromosome in fibroblasts") +
  #stat_cor(method = "pearson") +
  xlab("Number of Sex Chromosomes") +
  ylab("Total ZFX + ZFY TPM") +
  geom_smooth(color='grey', se=FALSE, method = lm) +
  theme_classic()
```


```{r}
#Plot total levels against chromosome count for lcl KDM
total_KDMl = data.frame(all_TPM_lcl$chrom_count, all_TPM_lcl$KDM)
ggplot(total_KDMl, aes(x =all_TPM_lcl.chrom_count, y =all_TPM_lcl.KDM)) +
  geom_point(color = "black") + 
  labs(title = "KDM6A + UTY expression vs sex chromosome in fibroblasts") +
  #stat_cor(method = "pearson") +
  xlab("Number of Sex Chromosomes") +
  ylab("Total KDM6A + UTY TPM") +
  geom_smooth(color='grey', se=FALSE, method = lm) +
  theme_classic()
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
#Overlap aneuploid responsive genes and DDX3X KD genes
#read in genes that are significantly differently expressed
malexkdsig = read.delim("/lab/solexa_page/shruthir/DDX3KD_RNAseq/output7doublekdtrimmed/maleControlvsDDX3X_kd_DESeq_exp_p0.05.txt")
```

```{r}
#Get X responsive genes
xresponse = read.delim("/lab/page/adrianna/RNA-seq_to_share/20221021/Autosome_log2FoldChange_per_X_or_Y_calculations/lcl_fib_XandY_results_auto_expressed.txt")
```

```{r}
#get significant X responsive genes
xresponsesig = xresponse[which(xresponse$fib_padj_X<0.05),]
```

```{r}
#Get overlap 
xresponseddx3x = malexkdsig[which(malexkdsig$Gene %in% xresponsesig$Gene),]
```

```{r}
#Generate Venn Diagram
library(ggplot2)
library(ggVennDiagram)

# List of items
x <- list(A = xresponsesig$Gene, B = malexkdsig$Gene)

# 2D Venn diagram
ggVennDiagram(x)
```

```{r}
#Get ZFX and ZFY CRISPRi TPM
ZFX_CRISPRi = read.delim("/lab/page/adrianna/RNA-seq_to_share/20230104/ZFX_ZFY_CRISPRi/normCounts_XY_singleKDs.txt")
ZFX_CRISPRi$Gene = row.names(ZFX_CRISPRi)
ZF = c("ZFX","ZFY","DDX3X","DDX3Y")
ZFXY = ZFX_CRISPRi[which(ZFX_CRISPRi$Gene %in% ZF),]
```

```{r}
write.table(ZFXY, file="/lab/solexa_page/shruthir/Adrianna/ZFXYKD_normcounts.txt", sep="\t", quote=F,col.names = NA)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
