---
title: "CRISPRicode"
output: html_document
date: "2023-04-13"
---
```{r}
knitr::opts_knit$set(root.dir = "/lab/solexa_page/shruthir/DDX3KD_RNAseq/output7doublekdtrimmed")
source("/lab/solexa_page/shruthir/Adrianna/ASR_functions_20190917_SR.R")
```


```{r setup, include=FALSE}
#Figure 3 and 4
#Metadata is a file containing 
metadata_limited = read.delim("metadata_DDX3X_Y copy.txt")
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
metadata_limited$cell_line <- as.factor(metadata_limited$cell_line)
#Make file table:
 annofile <- read.delim("/lab/page/bowtie-indexes/gencode.v24.annotation.basic_ccds_nopar.gene_tx_annotable.txt",stringsAsFactors = FALSE)
 tx2gene <- data.frame("TXNAME"=annofile$transcript_id, "GENEID"=annofile$gene_name)
 library(tximport)
crispri_files <- paste0(metadata_limited$kallisto_path) 
names(crispri_files) <- metadata_limited$sample

txi_ddx3 <- tximport(crispri_files, type = "kallisto", tx2gene = tx2gene)
 save(txi_ddx3,file="/lab/solexa_page/shruthir/DDX3KD_RNAseq/output7doublekdtrimmed/txi_ddx3_final.rda")
 load(file ="/lab/solexa_page/shruthir/DDX3KD_RNAseq/output7doublekdtrimmed/txi_ddx3_final.rda")

```
```{r}

library(DESeq2)
dds_ddx3 <- DESeqDataSetFromTximport(txi_ddx3, colData = metadata_limited, design = ~ genotype + cell_line)
 dds_ddx3 <- DESeq(dds_ddx3)
 save(dds_ddx3, file="/lab/solexa_page/shruthir/DDX3KD_RNAseq/output7doublekdtrimmed/dds_ddx3_final.rda")
load(file="/lab/solexa_page/shruthir/DDX3KD_RNAseq/output7doublekdtrimmed/dds_ddx3_final.rda")

```


```{r}
#run clustering and pca on raw results
vst_ddx3 <- vst(dds_ddx3, blind=FALSE)
sampleDists <- dist(t(assay(vst_ddx3)))

library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vst_ddx3$genotype, vst_ddx3$cell_line, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```

```{r}
library(ggplot2)
data <- plotPCA(vst_ddx3, intgroup=c("genotype", "cell_line"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=cell_line, shape=genotype)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```


```{r}
#Expressed gene list
metadata <- read.delim("/lab/solexa_page/shruthir/DDX3KD_RNAseq/output7doublekdtrimmed/metadata_DDX3X_Y.txt", stringsAsFactors = FALSE)

metadata$cell_line <- as.factor(metadata$cell_line)
metadata$batch_libprep = as.factor(metadata$batch_libprep)
metadata_female <- metadata[metadata$karyotype == "XX",]
metadata_male <- metadata[metadata$karyotype == "XY",]


load(file ="/lab/solexa_page/shruthir/DDX3KD_RNAseq/output7doublekdtrimmed/txi_ddx3_crispri.rda")
expressedGenes <- tpm1_genes_xx_xy_median(tximport_file = txi_ddx3_crispri, XX_samples = metadata_female$sample, XY_samples = metadata_male$sample)
 save(expressedGenes, file="expressedGenes.rda")
load(file="expressedGenes.rda")
```




```{r}
setwd("/lab/solexa_page/shruthir/Paperfigures/Scratch/")
myList <- matrix(c("DDX3X_kd","Control","DDX3Y_kd","Control"),ncol=2, byrow = TRUE)
deseqFactorAnalysis(deseqObject = dds_ddx3, myVariable = "genotype", expressedGeneList = expressedGenes$expressedGenes, comparisonList = myList, p_value=0.05, myTitle = "male")
```

```{r}
xyddx3xkd = read.delim("/lab/solexa_page/shruthir/Paperfigures/txtfiles/maleDDX3X_kdvsControl_DESeq_expressed.txt")
```


```{r}
  if (!requireNamespace('BiocManager', quietly = TRUE))
    install.packages('BiocManager')

  BiocManager::install('EnhancedVolcano')
library(EnhancedVolcano)
  
EnhancedVolcano(xyddx3xkd,
    lab = rownames(xyddx3xkd),
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 0.05,
    FCcutoff = 0.001,
    col=c('grey', 'grey', 'grey', 'blue'))

```

```{r}
xyddx3ykd = read.delim("/lab/solexa_page/shruthir/Paperfigures/txtfiles/maleDDX3Y_kdvsControl_DESeq_expressed.txt")
EnhancedVolcano(xyddx3ykd,
    lab = rownames(xyddx3ykd),
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 0.05,
    FCcutoff = 0.001,
    col=c('grey', 'grey', 'grey', 'blue'))

```
```{r}
#Remove DDX3X and plot
ddx3xkdremove <- xyddx3xkd[!(row.names(xyddx3xkd) == 'DDX3X'),]
EnhancedVolcano(ddx3xkdremove,
    lab = rownames(ddx3xkdremove),
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 0.05,
    FCcutoff = 0.001,
    col=c('grey', 'grey', 'grey', 'blue'))

```


```{r}
#Remove DDX3Y and plot
ddx3ykdremove <- xyddx3ykd[!(row.names(xyddx3ykd) == 'DDX3Y'),]
EnhancedVolcano(ddx3ykdremove,
    lab = rownames(ddx3ykdremove),
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 0.05,
    FCcutoff = 0.001,
    col=c('grey', 'grey', 'grey', 'blue'))
```

```{r}
#Extract ZFX and ZFY TPMs from Adrianna's DE-Seq object
load(file="/lab/solexa_page/shruthir/Adrianna/txi_zfx_zfy_skd_M.rda")
write.table(txi_zfx_zfy_M_skd$abundance, file="/lab/solexa_page/shruthir/Adrianna/zfxkdtpm.txt", sep="\t", quote=F,col.names = NA)

```






